forward/forwarder_test.go

```diff
@@ -188,6 +293,85 @@ func (s *ForwarderTestSuite) TestRequestNoReroutes() {
 	s.Error(err)
 }
 
+func (s *ForwarderTestSuite) TestRegisterListener() {
+	listener := &EventListener{}
+	listener.On("HandleEvent").Return()
+
+	s.forwarder.RegisterListener(listener)
+	s.Assertions.Equal(1, len(s.forwarder.listeners), "Expected 1 listener to be registered")
+
+	// remove all listeners
+	s.forwarder.listeners = nil
+}
+
+func (s *ForwarderTestSuite) TestEmit() {
+	// wait for HandleEvent being called
+	var wg sync.WaitGroup
+	wg.Add(1) // expect 1 call to HandleEvent
+
+	listener := &EventListener{}
+	listener.On("HandleEvent", mock.Anything).Run(func(args mock.Arguments) {
+		wg.Done()
+	}).Return()
+
+	s.forwarder.RegisterListener(listener)
+
+	// emit an empty struct
+	s.forwarder.emit(struct{}{})
```

Comment:
This test validates that the event emitting works as expected. I added it to cover the lines in emit. 

Preferably we should move the `emit` and `RegisterListener` to it's own package and implementation that one can embed in their own types. This would allow use to reuse the paradigm easier without duplicating it all over the place. For now I want to make sure that the emitting actually emits to a registered listener.